src/visip/dev/dtype_new.py

```diff
@@ -0,0 +1,582 @@
+            if b:
+                return True, _vm_merge(var_map, vm)
+
+    # if type is Int, Float, Str, NoneType, typeinfo must be the same
+    elif isinstance(type, (Int, Float, Str, NoneType)):
+        if typeinfo is type:
+            return True, var_map
+
+    # if type is Bool, typeinfo must be Bool or Int
+    elif isinstance(type, Bool):
+        if isinstance(typeinfo, (Bool, Int)):
+            return True, var_map
+
+    # if type is Class, typeinfo must be Class and type.origin_type must be subclass of typeinfo.origin_type
+    elif isinstance(type, Class):
+        if isinstance(typeinfo, Class) and issubclass(type.origin_type, typeinfo.origin_type):
+            return True, var_map
+
+    # if type is Enum, typeinfo must be Enum and type.origin_type must be subclass of typeinfo.origin_type or typeinfo must be Int
+    elif isinstance(type, Enum):
+        if isinstance(typeinfo, Enum) and issubclass(type.origin_type, typeinfo.origin_type) \
+                or isinstance(typeinfo, Int):
+            return True, var_map
+
+    # if type is Tuple, typeinfo must be Tuple and all args must be subtype
+    elif isinstance(type, Tuple):
+        if isinstance(typeinfo, Tuple) and len(type.args) == len(typeinfo.args):
+            for arg, arginfo in zip(type.args, typeinfo.args):
+                b, vm = is_subtype_map(arg, arginfo, var_map, const)
+                if not b:
+                    return False, {}
+                var_map = _vm_merge(var_map, vm)
+            return True, var_map
+
+    # if type is List, typeinfo must be List and args must be subtype
+    elif isinstance(type, List):
+        if isinstance(typeinfo, List):
+            b, vm = is_subtype_map(type.arg, typeinfo.arg, var_map, const)
+            if b:
+                return True, _vm_merge(var_map, vm)
+
+    # if type is Dict, typeinfo must be Dict and keys, values must be subtype
+    elif isinstance(type, Dict):
+        if isinstance(typeinfo, Dict):
+            b, vm = is_subtype_map(type.key, typeinfo.key, var_map, const)
+            if b:
+                var_map = _vm_merge(var_map, vm)
+                b, vm = is_subtype_map(type.value, typeinfo.value, var_map, const)
+                if b:
+                    var_map = _vm_merge(var_map, vm)
+                    return True, var_map
+
+    return False, {}
+
+
+def _vm_merge(a, b):
+    var_map = a.copy()
+    for t in b:
+        if t in var_map:
+            var_map[t] = Union([var_map[t], b[t]])
+        else:
+            var_map[t] = Union([b[t]])
+    return var_map
+
+
+class TypeInspector:
+    def is_constant(self, type):
+        return isinstance(type, Const)
+
+    def constant_type(self, type):
+        return type.arg
+
+
+def extract_type_var(type):
+    """
+    Returns set of all TypeVars from composed type.
+    :param type:
+    :return:
+    """
+    ret = set()
+
+    # TypeVar
+    if isinstance(type, TypeVar):
```

Comment:
Use private methods in the instances.

Reply:
could be even public method `typevar_list`, e.g.:
```
vars = type_hint.typevar_list()
```

Reply:
Hotovo.